crates/relayer-framework-test/src/relayer_mock/contexts/chain.rs

```diff
@@ -0,0 +1,236 @@
+//! The MockChainContext uses a ChainState alias for it's consensus_state.
+//! This type is a HashMap<Height, State>, used to represent the State of
+//! the chain at all Heights.
+//! The consensus_states is a HashMap<Height, ChainState> used to send the
+//! consensus_state of the chain at a specific Height inside a client upgrade
+//! message.
+//! The client_states is a HashMap<ClientId, HashMap<Height, ChainState>>.
+//! This is used to check the consensus_state for a specific client, at a
+//! specific height.
+//! Usually the client_states would use the root hashes of a Merle Tree,
+//! but since the MockChain is used for testing and will have a small number
+//! of states, the whole state is used. This avoids needing to implement
+//! Merkle Trees and Proofs.
+
+use alloc::string::String;
+use std::collections::hash_map::Entry;
+use std::sync::Mutex;
+use std::{collections::HashMap, sync::Arc};
+
+use crate::relayer_mock::base::error::Error;
+use crate::relayer_mock::base::types::aliases::{
+    ChainState, ChannelId, ClientId, ConsensusState, PortId, Sequence,
+};
+use crate::relayer_mock::base::types::runtime::{MockChainRuntimeContext, MockRuntimeContext};
+use crate::relayer_mock::base::types::{
+    height::Height as MockHeight, packet::PacketKey, state::State,
+};
+use ibc_relayer_framework::base::one_for_all::traits::runtime::OfaRuntimeContext;
+
+pub struct MockChainContext {
+    pub name: String,
+    pub consensus_states: Arc<Mutex<HashMap<MockHeight, ChainState>>>,
+    pub consensus_state: Arc<Mutex<ChainState>>,
+    pub client_states: Arc<Mutex<HashMap<ClientId, HashMap<MockHeight, ChainState>>>>,
```

Comment:
This is actually the `consensus_states` we talked about. The `client_states` field should be something like:

```rust
client_states: Arc<Mutex<HashMap<ClientId, ClientState>>>
```

Where `ClientState` is a different type that contains information such as the client type and whether the client has been frozen. However since we do not model that part at the moment, we can skip defining the actual `client_states` field.